### DISCUSSION TITLE
General UI improvements

### DISCUSSION THREAD

**Discussion by evanheller** [zed-industries/zed#20086] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

It might be good to open some discussion about general UI improvements to better utilize screen real-estate and functionality, building on #19537.

1) The title bar is really very big, and even though it gets used for some chat/call icons (when those features are being used), it feels like a very inefficient use of space, especially in combination with the status and tab bars. 

<img width="967" alt="titlebar_size" src="https://github.com/user-attachments/assets/a32131bd-73d9-41fa-a50c-2290ca005ab2">

&nbsp; 
What could one do about this? Some possibilites for thought:

A) Some functionality could be shifted from the status bar to the title bar. The quick action buttons certainly could make sense there. Some buttons in the tab bar also could make sense, such as splitting and full-screening, since those features are not specific to the active tab.

B) The status bar itself seems like it could go into the title bar.

C) The title bar can gain additional, project-related functionality, where each project gets its own tab in the title bar. Currently, only one project can be open in a given window of Zed, but it's not too crazy to think you might jump between multiple projects.

&nbsp; 

2) Related to the above, the placement of tab bar and status bar buttons feels awkward, particularly when both are being used at the same time. Some re-thinking of the interface might help.

<img width="1035" alt="buttons" src="https://github.com/user-attachments/assets/890566d2-79f9-4633-8550-21b837383345">

&nbsp;

3) Collaboration features, which are not specific to a given project, are
grouped with project-specific features (project layout, outline, and project diagnosis) in the left-side panel. The chat panel, whose functionality is linked to the collaboration panel, however, is on the right-side panel. Thematically, this grouping doesn't make much sense, and the left/right panel placement of two things that are very closely connected feels awkward.

<img width="1027" alt="collaboration_panels" src="https://github.com/user-attachments/assets/50868d84-dcb2-415c-9271-a30e665babd2">

&nbsp; 
Probably a more UI/UX-centric person can be more helpful, but to me it feels like collaboration features (both the channel list and chat) belong in the right side panel.

**Comment by JosephTLyons** [comment]
I'm going to move this into a discussion, as that seems like a better place to organize thoughts on this.

**Comment by jacobwhall** [comment]
Here's what the top bar looks like for me (with gutters off, UI and Buffer font sizes set to 20):

![image](https://github.com/user-attachments/assets/23117bc4-717f-49d5-90fe-004916fc6ee0)

In my humble opinion, the top bar looks great. It isn't very large compared to the rest of the UI, and its clean look helps me maintain focus on my project. Even with the window filling half a laptop screen, I don't ever feel like my work is being crowded out by the top bar.

I do think that those status bar buttons are getting crowded, especially with the new REPL feature (which works great, btw!). While I understand *why* there are tab bar buttons separated from the status bar buttons, in practice they turn into a pile of icons along with the minimize/maximize/close window buttons and GitHub login dropdown.

In my ideal world, here's what would happen:

- editor controls get moved to the workspace options in the hamburger menu on the top-left of the window. I rarely touch these and even more rarely need to control them per-tab.
- inline assist gets moved to the gutter, next to (or instead of) the lightning bolt. this would make more sense for an inline tool; I often click on that icon expecting to get the AI assistant panel (which confusingly uses the SAME icon at the bottom the screen!)

This would leave selection controls and search at the top, as well as the REPL if it's enabled. I think this would be a lot cleaner and fit better beneath the tab bar buttons. I could see the REPL menu getting moved either to the gutter with inline assist, or maybe even down to the bottom next to the terminal button. However, I kinda like it where it is right now (though its spacing on the right could be improved)

I haven't used the collaboration features much so I won't comment much on that. In general, it seems a bit awkward to put social channels inside editor windows, though ya'll have done a pretty neat job of it so far.

Hope this helps, and thanks for your work on Zed. I've really been enjoying using it.

**Reply by jayliaoo** [reply]
It works fine on the desktop, but not great on laptops.

**Reply by zedmakesense** [reply]
I think you can remove most of the status bar stuff, I have removed some (or maybe all?) -
<img width="1366" height="768" alt="image" src="https://github.com/user-attachments/assets/31cf11ca-2369-443b-b31d-bc3410027e61" />

**Comment by ghost** [comment]
i've pointed out many problems with the zed ui numerous times in numerous places throughout the project, be it in issues, discussions, on the discord channel, etc... but it seems to go unheard so i gave up. i just came across this, and decided to give it a last chance hoping for anyone in the zed team to listen for once. zed is a bit secretive, we don't really know if someone in the team is prototyping and working on making the zed ui better as we speak, but in its current state i have to bring myself to say it, it's god damn awful. zeds ui is retro, it's ancient, it's not thought through. a lot of wasted space like the top bar, huge paddings, a lot of inconsistencies in buttons and font sizes, the list just continues... worst of them all imo is the layouts and widgets, i have no idea why the zed team thought creating a ui framework that looks as non-native as possible in the modern day where native is adored was a good idea.

Edit: Here is a more detailed message i sent on the [discord channel](https://discord.com/channels/869392257814519848/1255246362220691456/1271156098203648113)
> 1. font sizes are inconsistent throughout the editor, f.e compare the file explorer and the chat panel.
> 2. unnecessarily large paddings to many elements, like the gutter and the extensions panel.
> 3. a ton of wasted ui space in favor of the "collaborative core of zed" which i could bet a majority of users wont be using zed for most of their time, like the top/title bar that is used to display users on a call, could instead be used to visualize tasks (like xcode) or search (vscode).
> 4. the layout of views throughout the editor are fixed, no option to customize them to ones liking, f.e i dont like the bottom bar but hiding it would remove access to all the relevant buttons. in vscode, you can hack the layout of the views in almost any way you want either via setting options, extensions or raw css customizations. even xcode, a closed source free but proprietary ide with a very opinionated ui/ux even does this better than zed.
> 5. no native ui, which is fine to an extent, but doesnt conform to apples hig in the slightest for the macos version, just native tooltips and equivalent changes would make a huge improvement if you ask me, vscode and most other gui editors does this and can achieve a more modern look, and they do this because people want this... which seems to me zed have rejected multiple times in favor of the "retro fan" minority that can be seen in this github discussion https://github.com/zed-industries/zed/discussions/8499 (fyi i love the figma concept someone posted in the comments)
> 6. a lot of the default themes have incorrect color values and applies colors to incorrect tokens, there are contrast issues in a lot of the themes... someone decided to modify them and now refuse changes to correct them.
> 7. worst of them all imo.. a big "sign in" button that is never out of your view and that you will never be able to remove without having to login with your github account, reminder the zed editor is foss... even xcode, a proprietary ide, doesnt clug their ui with "sign in" for you to use their services like xcode cloud..

**Reply by phisch** [reply]
I think this is quite some over-reaction. To me, a lot of the things in zed look fantastic. It has some areas that need to be improved, but it's still early development. You can't expect everything to be perfect from day 1.

![image](https://github.com/user-attachments/assets/35d749df-4f68-4cfc-a5ae-186b1483ed0a)

As for the top bar, you guys keep saying it is underutilized, but its space is currently used for code share participants. I remember reading in an issue that the zed team would be open to add a "hide titlebar" feature, if there comes a better UI for those participant buttons.

**Reply by ghost** [reply]
<img width="1470" alt="Skärmavbild 2024-11-01 kl  20 39 21" src="https://github.com/user-attachments/assets/bc3fdd0a-3064-43bb-94e2-47f6704a2476">
all i see is wasted space when i dont do collaboration. and no, not an over reaction, the ui is plain awful on macos. the layout looks better on linux because that's how many linux apps looks like in general, but this is not how macos apps looks like (except awful electron apps).

**Reply by ghost** [reply]
you're also talking as if zed is completely new, yes it's new on linux. but it's been on macos for years, and none of these major ui/ux problems have been addressed. if you ask me, the macos version needs a complete redesign, one that's been thought through carefully. software on apple platforms are all about quality and user experience, zed doesn't provide that unfortunately. if you're gonna argue back at that, just a heads up, zed doesn't even conform to apples HIG the slightest.

**Reply by ghost** [reply]
react native, despite not a native framework, is doing it's best to provide near native experience as possible, there are so much efforts of it trying to mimic the capabilities of swiftui on apple platforms (they still lack behind a lot), and there are reasons for that, both because it looks better, elegant, modern, and goes in line with the software expectation on said platform, and because users love it. zed looks to be doing the complete opposite, the ui/ux is targeting a minority that are okay with retro looking software, likely coming from linux and android (not that there is anything wrong with that). but this is just not acceptable on macos if you ask me.

**Reply by jacobwhall** [reply]
> zed is a bit secretive, we don't really know if someone in the team is prototyping and working on making the zed ui better as we speak

This is an odd claim. Zed is one of the fastest-moving development projects I've ever encountered. They release updates nearly every day, and you are free to browse their commit history in this repository. For example, the commit history for the gpui crate can be found [here](https://github.com/zed-industries/zed/commits/main/crates/gpui). I don't know why you feel entitled to know what Zed developers are working on next: they are releasing free software, on their own schedule, with their own money. You will always be able to contribute code yourself (assuming the Zed team likes your vision, and you agree to their CLA), or fork the project and make out of it what you want.

> react native, despite not a native framework, is doing it's best to provide near native experience as possible...zed looks to be doing the complete opposite

> the ui is plain awful on macos

> zeds ui is retro, it's ancient, it's not thought through

> i have no idea why the zed team thought creating a ui framework that looks as non-native as possible in the modern day where native is adored was a good idea

Yes, Zed is clearly not prioritizing a native MacOS feel. It's fair to want it to fit in better among your other MacOS apps but I feel it's disingenuous to describe Zed as "the opposite [of] elegant, modern." I can't help but think you're being intentionally inflammatory. The Zed devs have clearly put a lot of careful effort into the look and feel of this app (and their website) in a relatively short time.

> the ui/ux is targeting a minority that are okay with retro looking software, likely coming from linux and android (not that there is anything wrong with that). but this is just not acceptable on macos if you ask me

> in its current state i have to bring myself to say it, it's god damn awful

It honestly just sounds like you'd prefer to use a different editor.

I agree with some of your points, but it's hard to take you seriously. It is not my responsibility to enforce it, but I recommend you take a minute to review [Zed's code of conduct](https://zed.dev/code-of-conduct).

**Reply by ghost** [reply]
> This is an odd claim. Zed is one of the fastest-moving development projects I've ever encountered. They release updates nearly every day, and you are free to browse their commit history in this repository. For example, the commit history for the gpui crate can be found [here](https://github.com/zed-industries/zed/commits/main/crates/gpui). I don't know why you feel entitled to know what Zed developers are working on next: they are releasing free software, on their own schedule, with their own money. You will always be able to contribute code yourself (assuming the Zed team likes your vision, and you agree to their CLA), or fork the project and make out of it what you want.

That's true and i'm aware, it's an open source project after all. However, i don't see how that makes it less important for communication with the open source community that tries to contribute to zed, see https://github.com/zed-industries/zed/discussions/20068

> Yes, Zed is clearly not prioritizing a native MacOS feel.

And my point is precisely that that's a mistake.

> It's fair to want it to fit in better among your other MacOS apps but I feel it's disingenuous to describe Zed as "the opposite [of] elegant, modern." I can't help but think you're being intentionally inflammatory. The Zed devs have clearly put a lot of careful effort into the look and feel of this app (and their website) in a relatively short time.

The website is fine, i don't think there have been a lot of "careful effort into the look and feel of this app" given how many inconsistent parts there are throughout the app and how much screen estate is being wasted.

**Reply by imbev** [reply]
> > Yes, Zed is clearly not prioritizing a native MacOS feel.

> And my point is precisely that that's a mistake.

A MacOS design will feel less native on Linux

**Reply by phisch** [reply]
To be fair, it doesn't feel "native" on linux either. It feels like its own thing, and that's good imo.

**Reply by evanheller** [reply]
I would say that how well Zed fits in on MacOS (or any particular OS) is a sub-point to a larger set of suggestions/critiques/criticisms of Zed's UI aimed at improving its usability, efficiency, consistency, as well as overall look and feel ("native UI" was indeed only point 5 of @nervenes reply). It's a valid one, but not necessarily the main one. I would love to see more UI prototypes from those with suggestions (kind of like those that appeared in #8499). Maybe a really compelling mockup would do the trick.

And, it can't hurt to be said: the point is not to only pile on complaints and criticisms. There's a lot that's compelling in Zed or we wouldn't be here. The point is that when you sit down to use it for your work, you notice some things that are awkward or a little clunky that might eventually mean the difference between using Zed or forgetting about it and sticking to our existing tools (and we're rooting for them!)

**Comment by garfunkel** [comment]
The point about it not having a particular OS's look and feel is moot. If it did then there would be no consistency with Zed on other platforms. Many users use multiple platforms and they don't want a different experience with different limitations/constraints on each just to get work done. You either have it inconsistent with other apps, or you have it inconsistent with itself. Pick one. I'd much rather the former, and it's less work too.

My own view is that Zed looks pretty good now, but I agree that the amount of chrome vertical height should be reduced, or bars combined somehow to achieve it. Ultimately having something as configurable as possible (where buttons can be moved etc) is the best bet to make everybody happy. I mostly agree with the OP but I think we need some mockups so we can visualise things.

**Comment by xcrap** [comment]
<img width="2730" alt="Screenshot 2024-11-02 at 07 27 23" src="https://github.com/user-attachments/assets/b12ba041-f0f4-4ee5-8c8a-d822eef83a55">

This is how it looks here, I actually think the distribution is great and if there's anything missing it's just tiny icon spacing adjustments. The heights feel very good, there's nothing to shrink here in my opinion.

**Reply by ghost** [reply]
I could see how you feel that on such a large monitor, but consider us with 13" 14" MacBooks. The wasted screen estate and the large paddings here and there is very annoying. And not to say, you have removed the breadcrumbs and the gutter line numbers in the screenshot...

**Reply by xcrap** [reply]
> I could see how you feel that on such a large monitor, but consider us with 13" 14" MacBooks. The wasted screen estate and the large paddings here and there is very annoying. And not to say, you have removed the breadcrumbs and the gutter line numbers in the screenshot...

Yes I use a more minimal version, I also have a 14" Macbook and it's fine, my point is about the height of the elements like the titlebar seems actually bigger on your screenshot but maybe it's just a feeling from my part. But the meaning is some people prefer more relaxed spacing between elements giving them space to breath. I understand it can be too big in your case but you can also close tabs and access files in different ways or maybe could even be the selected typography, i feel mine has a smaller gap, have you tried to tweak with your ui_font_size ? It looks bigger than the buffer font size.

**Reply by ghost** [reply]
My ui font size is set to 15 and my buffer font size is set to 12, and that's to get roughly the same font size:
<img width="1566" alt="Skärmavbild 2024-11-02 kl  10 15 04" src="https://github.com/user-attachments/assets/257f660d-4c77-40c3-862b-23771bdebcf9">

With them both set to 12, the ui font size is incredibly small:
<img width="1566" alt="Skärmavbild 2024-11-02 kl  10 15 39" src="https://github.com/user-attachments/assets/3b8efaa9-954d-411b-89b3-6c69856e0aa2">

Again, these are one of the inconsistencies issue i mentioned in my bullet list.

**Reply by xcrap** [reply]
> With them both set to 12, the ui font size is incredibly small:

Small, but so beautiful :) 

I think there will be many opinions here, for me when I'm coding I even remove the navigation dock and just use the cmd p to access files, but the ui font makes a huge difference on those titles vertical spacing, maybe they could have ui fonts sizes for titles and ui font sizes for the remaining of the ui. I've always loved the look of Atom and I think Zed is actually better, with that small exception of the icon placements :)

**Reply by ghost** [reply]
Small, but barely readable. It's too small on my 13" :/ That 12 font size isn't actually 12, that has to be 8-9. I hope you can see that from the average user perspective, this isn't any good.

**Reply by xcrap** [reply]
> Small, but barely readable. It's too small on my 13" :/ That 12 font size isn't actually 12, that has to be 8-9. I hope you can see that from the average user experience, this isn't any good.

Like I've mentioned before, I think this will be very subjective in the sense that some people prefer to see the code bigger and the interface smaller or vice versa, I never used VIM but compared with other editors that I've used, Sublime, PHPStorm and VSCode which i still use occasionally Zed has the most beautiful polished interface :) I don't mind some of the things moving to the titlebar, as long as it's not overloaded... or that we can't disable, it would be fine by me. Enjoy the weekend :)

**Comment by ghost** [comment]
> The point about it not having a particular OS's look and feel is moot. If it did then there would be no consistency with Zed on other platforms.

> A MacOS design will feel less native on Linux

The linux and windows app already have different layouts compared to the macOS version, to some extent. See the left side bar and the title bar. So I don't see how that's an argument, given there are already os dependent ui implementations.

**Comment by xcrap** [comment]
But I do agree with the set of those two rows of icons, that's why I've disabled the breadcrumbs and quick_actions.

**Comment by garfunkel** [comment]
Whatever you do, just don't add anything like that stupid left-hand-side vertical button bar thing in VSCode.

**Comment by ghost** [comment]
As for the side bars, status bar and title bar layouts, this is how zed is currently constructed:
<img width="753" alt="Skärmavbild 2024-11-02 kl  14 54 12" src="https://github.com/user-attachments/assets/7519926d-1014-4f9e-91ff-279abe293b2a">
<img width="753" alt="Skärmavbild 2024-11-02 kl  14 54 16" src="https://github.com/user-attachments/assets/c518c3e8-5d61-4a98-8319-816a5051c76d">

But ideally, this is how it should be:
<img width="753" alt="Skärmavbild 2024-11-02 kl  14 57 09" src="https://github.com/user-attachments/assets/7cd5bc92-dc45-4d4e-b296-3a982248fba0">
<img width="753" alt="Skärmavbild 2024-11-02 kl  14 58 31" src="https://github.com/user-attachments/assets/2aa82476-2124-4247-850d-f6e0ded603b2">

that is, for macOS applications. Linux and windows seems fine as it is.

**Reply by jackTabsCode** [reply]
I agree and after seeing this, it's absolutely how most macOS apps behave. Not sure why people are downvoting. Nice catch.

**Reply by LYZ02** [reply]
Zed is not specifically designed for macOS, meaning Zed does not need to comply with macOS development standards. However, even without adhering to such guidelines, I still think Zed's interface design is excellent.

**Reply by ghost** [reply]
They develop an application for macOS though, that on itself implies some standards to follows, of which is how macOS apps should behave and to an extent look like. You wouldn't really create a Windows XP like app today and expect the average users to pay for that on Windows 11, the only software that gets away with that are niche software that actually started in Windows XP and have became an industry standard somewhere.

**Comment by xcrap** [comment]
Don't agree, the title bar should not change according to the left or right docks windows, ever!!!

**Comment by ghost** [comment]
well, just to make it clear, the freeform sketch wasn't a personal opinion, that's how the native apple frameworks implement the layout by default, that's the layout most macos ui frameworks try to follow, that's the layout most modern apps try to conform to, that's the layout apple is endorsing in their HIG. if you don't like it then maybe you might want to look for another operating system. the current zed layout should be second class on macos as a "custom" layout, while the proper style should be the default. stale title bars are not home on modern macos, keep that to windows and linux.

**Comment by sangelxyz** [comment]
I would like to be able to disable it in settings, takes up far to much space.

**Comment by BillyDM** [comment]
IMO the biggest thing needs reworked is the breadcrumbs bar (or whatever it is you call the bar directly underneath the tabs). It uses up way more precious vertical space than it should.

I also don' think that search/inline-assist/selection/controls toolbar buttons should be on the breadcrumbs bar, they should be moved somewhere else.

IMO VSCode already has the height of the breadcrumbs bar nailed.

**Comment by Andriamanitra** [comment]
Zed UI does look and feel kind of clunky compared to other more established modern editors like Sublime and VS Code (pictured below). I have a few ideas that might help:
* Come up with a clear strategy for where buttons should be located. Currently they are sprinkled all over the interface with seemingly no logic behind it. Why is project diagnostics in the bottom left even though it opens a new tab in the editor (rather than the sidebar it's located next to)? Why are there two similar dropdown menus with two different icons on opposite sides of the top bar? Why are the social features split between three different corners of the editor? Etc.
* Make the buttons take the full height of the bar they're inside (both VS Code and Sublime do this). There's no good reason to have such tiny buttons with so much empty space around them, it just makes it harder to hit them with the mouse. This is a problem especially in the bottom bar.
* Get rid of (or soften) some of the borders. Too many lines between elements makes the interface look more cluttered than it actually is (compare with Sublime and VS Code in the screenshot below).
* Use fewer inscrutable icons. I doubt I could guess what even half the buttons visible by default do without hovering over them to see the tooltip. Not every feature needs a dedicated button – a lot of the less commonly used functionality can live in the command palette.

[![screenshot of sublime, zed, and VS code](https://github.com/user-attachments/assets/81143785-6309-4885-be14-37db005d6522)](https://github.com/user-attachments/assets/81143785-6309-4885-be14-37db005d6522)

**Comment by waynehoover** [comment]
It would be such a standout feature to move tabs into the title bar like browsers have been doing for ages. This seems like a no brainer, otherwise we have a whole row of precious vertical space just wasted.

**Comment by VKondakoff** [comment]
This! 👆

<img width="1386" alt="Снимок экрана 2024-11-02 в 23 50 55" src="https://github.com/user-attachments/assets/06da15f8-8a0c-4982-abe8-419385d858fc">

**Reply by ghost** [reply]
I wouldn’t oppose, as it would go great with the macos layout i showed in above posts.

**Reply by imbev** [reply]
The placement of tabs within CSD may be a more efficient use of space, but it harms usability IMO.

I would probably switch to a different editor over that alone.

**Reply by waynehoover** [reply]
It could be a setting.

**Reply by vizh** [reply]
How can we visualize horizontal and vertical splits in this case? This will work only if they are not present.

**Reply by jackTabsCode** [reply]
Yep, and it's even a setting in Safari (made by Apple, who notoriously does not have a lot of settings)

**Comment by Cheirnomus** [comment]
A run / play button that is actually a task trigger would be nice. When it is clicked on it reruns the last task picked and when it is shift-clicked it shows the task modal / picker menu.

The menu could also be filled with all the common tasks like running files or selections within a file, this would also fast track newcomers instead of having them getting bogged down with task configuration right out of the gate.

**Reply by RafatRashid** [reply]
This would probably be the best usage. 

<img width="638" height="235" alt="image" src="https://github.com/user-attachments/assets/a91c240f-3dfa-4608-993b-39f50941a6c9" />

The last run task can remain selected, with a list of all debug/task configs, that can be either run or debug. Image from GoLand.

**Comment by Spirarel** [comment]
I [feel strongly](https://www.reddit.com/r/ZedEditor/comments/1my5zav/ways_to_reduce_wasted_space/) about this issue. Has there been any developments here since last year?

**Reply by simonsolnes** [reply]
Some talk about redesign here: https://github.com/zed-industries/zed/pull/37428#issuecomment-3333544424

**Comment by xipeng-jin** [comment]
Hello Zed, I would like to share some adjustments for UI design in my own [fork](https://github.com/xipeng-jin/zed/tree/feature/minimal-ui). Since I am using a 14-inch MacBook for my daily driver and I love to use Zed editor so much ❤️, but the only thing annoying me is the large top title bar occupied lots of space in my vertical real estate. And personally I rarely touch any buttons in the title bar. Here is the screen recording for my version of implementation:

https://github.com/user-attachments/assets/570ee6e5-8347-45c0-b334-2cd1293bca2d

Here is the list for the new UI settings capability

1. Introduces a `title_bar.show` setting wired through the Title Bar module to dynamically attach or remove the native chrome.
2. Extends the windowing layer with APIs to vertically center macOS traffic lights and to toggle their visibility, letting the chrome float precisely where the condensed layout needs it when the title bar disappears.
3. Rethinks the left dock so it can reserve space for those controls, adds a native-height placeholder, and keeps the dock layout stable as panes open and close—no more traffic-light overlap in minimal mode as mentioned in this issue #38547 .
4. Links the new dock reservation to runtime state: when the title bar is hidden the dock now owns the traffic lights, shows them only when it’s open, and keeps them anchored to the tab stack.
5. Tightens the workspace toolbar with smaller padding, adaptive borders, and awareness of the tab bar’s visibility, trimming visual chrome so editors and terminals get every available vertical line.

Finally, really appreciate this PR #37428 to give me some initial inspirations for doing my own work on top of it. The reason why I didn't open a PR for it because I saw some comments in that thread about 

> there are plans for the new redesign to heavily use the title bar and the new redesign wants it to always be present

So I don't think merging my code for now is necessary for Zed roadmap. But I am happy to hear any feedback on it and it would be super great to inspire the future Zed UI design roadmap ☺️. Thanks for your great work Zed team!

**Comment by notnotjake** [comment]
I think we would be better off removing the status bar than the title bar. Removing the title bar introduces issues because there is not a good place to put macOS window controls with it gone and it makes it less obvious how to move the window (no large areas to click and drag from). 
The status bar however could move its content to a floating element inside the active pane. And panel buttons can be moved to the title bar (which would be more like other application UI's). This would free up vertical space with fewer compromises. 

This is what I've done in my fork of Zed and I really enjoy it. I also encourage anyone thinking about this stuff to just maintain your own fork! It's really easy and Zed can be crafted perfectly _for you_!

**Reply by skarndev** [reply]
There are users of tiling WMs on all major systems (including Mac and Win) which may not need a titlebar at all, as they typically close/minimize windows using the keyboard.
Aside from that, 
<img width="1512" height="286" alt="image" src="https://github.com/user-attachments/assets/3abc6090-1260-4700-b169-fdeaf444a0c1" />
Here is my redesign of VSCode that still allows trafficlights in a titlebar-less setup. It just occupies some space in the tabbar.
I do not see how such opt-ins would hurt the design considering they are config options.

**Comment by zedmakesense** [comment]
It would be nice to have an option to show some stats on the top title bar (in the middle) or maybe just the file you are currently on for minimalists.

<img width="1366" height="768" alt="image" src="https://github.com/user-attachments/assets/31cf11ca-2369-443b-b31d-bc3410027e61" />

Edit: added screenshot for some reason.

**Comment by zedmakesense** [comment]
Just saying but I dont think stuff should be changed cuz its the default, just give a toggleable option, if you dont like the default then just change the config.

**Comment by franciskafyi** [comment]
Thank you for everyone's feedback! I'm doing some spring cleaning 🧹 and I'm closing this discussion. Here's why:

- the original feedback is outdated
- catch all discussions like this are really hard for us to manage and triage properly
- we're very deliberate (and opinionated) about our design choices and we allow for customization when we think it makes sense

If you have specific ideas or input we always welcome it! Please open a new discussion that's well scoped around a singular problem area – and ideally with a proposed solution or examples as well 😄 ❤️
